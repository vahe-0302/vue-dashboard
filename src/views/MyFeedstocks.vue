<template>
  <div>
    <MyFeedstocksList
      :my-feedstocks="myFeedstocks"
      :to-select-feedstock="toSelectFeedstock"
    />
  </div>
</template>

<script>
import { mapState } from 'vuex'
import MyFeedstocksList from '@/components/MyFeedStocksList'

export default {
  components: {
    MyFeedstocksList
  },
  props: {
    toSelectFeedstock: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    ...mapState('feedstocks', ['myFeedstocks'])
  }
}
</script>

<style scoped></style>
