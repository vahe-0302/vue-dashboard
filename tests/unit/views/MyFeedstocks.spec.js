import Vuex from 'vuex'
import { cloneDeep } from 'lodash'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import MyFeedstocks from '@/views/myFeedstocks'

const defaultStoreStructure = {
  modules: {
    feedstocks: {
      namespaced: true,
      actions: {
        getMyFeedstocks() {
          setTimeout(() => {}, 10000)
        }
      },
      state: {
        myFeedstocks: [
          {
            id: 'apple_filter_cake',
            material: 'Apple filter cake',
            material_type: 'crop_residues',
            creator: 'admin',
            source: '',
            nitrogen: '1.2',
            carbon: '15.6',
            cn_ratio: '13',
            moisture_content: '60',
            bulk_density_yd: '1197',
            bulk_density_m: '710.15',
            public: true
          },
          {
            id: 'apple_pomace',
            material: 'Apple pomace',
            material_type: 'crop_residues',
            creator: 'other',
            source: '',
            nitrogen: '1.1',
            carbon: '52.8',
            cn_ratio: '48',
            moisture_content: '88',
            bulk_density_yd: '1559',
            bulk_density_m: '924.92',
            public: true
          }
        ]
      }
    }
  }
}

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

describe('MyFeedstocks', () => {
  let storeStructure
  beforeEach(() => {
    storeStructure = cloneDeep(defaultStoreStructure)
  })

  test('should render correctly', () => {
    const store = new Vuex.Store(storeStructure)
    const wrapper = shallowMount(MyFeedstocks, {
      store,
      localVue
    })
    expect(wrapper).toMatchSnapshot()
  })
})
