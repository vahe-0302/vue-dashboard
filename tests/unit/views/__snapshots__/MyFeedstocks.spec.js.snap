// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFeedstocks should render correctly 1`] = `
<div>
  <myfeedstockslist-stub></myfeedstockslist-stub>
</div>
`;
